contrib/mypy/src/python/pants/contrib/mypy/tasks/mypy_task.py

```diff
@@ -18,10 +19,22 @@
 from pex.interpreter import PythonInterpreter
 from pex.pex import PEX
 from pex.pex_info import PexInfo
+from pants.task.lint_task_mixin import LintTaskMixin
 
 
-class MypyTask(ResolveRequirementsTaskBase):
-  """Invoke the mypy static type analyzer for Python."""
+class MypyTaskError(TaskError):
+  """Indicates a TaskError from a failing MyPy run"""
+
+class MypyTask(LintTaskMixin, ResolveRequirementsTaskBase):
+  """Invoke the mypy static type analyzer for Python.
+
+  Mypy lint task filters out target_roots that are not properly tagged according to
+  --whitelisted-tag-name, and executes my py on targets in context from whitelisted target
+  roots. Next, if any transitive targets from the filtered roots are not whitelisted, a Mypy
+  TaskError will be thrown.
+
+  'In context' meaning in the sub-graph where a whitelisted target is the root
+  """
 
```

Comment:
To allow the task (options) to continue to be used in the previous location, you'll want to add:
```
deprecated_options_scope = 'mypy'
deprecated_options_scope_removal_version = '1.20.0.dev2'
```

See https://github.com/pantsbuild/pants/blob/570fab8d3896619b3005c542ee7e61afbd412e67/src/python/pants/option/optionable.py#L62-L67 for more info.